### PR TITLE
SONARPY-512 Fully qualified name for Function and Class definition symbols

### PR DESCRIPTION
**Pull request by andrea-guarino-sonarsource**

### CODE REVIEW

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSensor.java [hunk 1]
```diff
@@ -72,9 +78,31 @@ public void execute(SensorContext context) {
...
+  }
+  // visible for testing
+  static Map<URI, String> getInputFileByPackage(File projectBaseDir, List<InputFile> inputFiles) {
+    Map<URI, String> packageNamePerFile = new HashMap<>();
+    for (InputFile inputFile : inputFiles) {
+      String packageName = findInitFiles(inputFile, projectBaseDir);
+      packageNamePerFile.put(inputFile.uri(), packageName);
+    }
+    return packageNamePerFile;
+  }
```
Comment (pynicolas) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSensor.java]: Do we really need this method? Do we need to build a map for all InputFiles? What about moving `findInitFiles` to `PythonScanner` and call it inside `scanFile`?

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSensor.java [hunk 2]
```diff
@@ -72,9 +78,31 @@ public void execute(SensorContext context) {
...
+
+  private static String findInitFiles(InputFile inputFile, File projectBaseDir) {
```
Comment (pynicolas) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSensor.java]: What about renaming this method to something like `pythonPackageName`?

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSensor.java [hunk 3]
```diff
@@ -72,9 +78,31 @@ public void execute(SensorContext context) {
...
+    File currentDirectory = inputFile.file().getParentFile();
+    Deque<String> packages = new ArrayDeque<>();
+    while(!currentDirectory.getAbsolutePath().equals(projectBaseDir.getAbsolutePath())) {
+      File initFile = new File(currentDirectory, "__init__.py");
+      if (initFile.exists()) {
+        packages.push(currentDirectory.getName());
+      }
```
Comment (pynicolas) [inline sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSensor.java]: Shouldn't we `break` if `!initFile.exists()`?

File: sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSensor.java (hunk 1 above)
Comment (andrea-guarino-sonarsource) [reply sonar-python-plugin/src/main/java/org/sonar/plugins/python/PythonSensor.java]: Good point, I will do that

**Pull request merged**